pkg/loader/compose/v1v2.go

```diff
@@ -268,12 +269,154 @@ func libComposeToKomposeMapping(composeObject *project.Project) (kobject.Kompose
 			log.Infof("Service name in docker-compose has been changed from %q to %q", name, normalizeServiceNames(name))
 		}
 	}
+
+	// This will handle volume at earlier stage itself, it will resolves problems occurred due to `volumes_from` key
+	handleVolume(&komposeObject)
+
 	return komposeObject, nil
 }
 
+// This function will retrieve volumes for each service, as well as it will parse volume information and store it in Volumes struct
+func handleVolume(komposeObject *kobject.KomposeObject) {
+	for name, _ := range komposeObject.ServiceConfigs {
+		// retrieve volumes of service
+		vols := retrieveVolume(name, *komposeObject)
+		var FinalVol []kobject.Volumes
+
+		// Check for duplication of volumes
+		for _, v := range vols {
+			if !findDuplicateVol(v, FinalVol) {
+				FinalVol = append(FinalVol, v)
+			}
+		}
+		var count int
+
+		// Parse volume information and store it in Volumes struct
+		for v, _ := range FinalVol {
+			vol := &FinalVol[v]
+			var err error
+			vol.VolumeName, vol.Host, vol.Container, vol.Mode, err = transformer.ParseVolume(vol.MountPath)
+			if err != nil {
+				log.Warningf("Failed to configure container volume: %v", err)
+				continue
+			}
+			log.Debug("Volume name %s", vol.VolumeName)
+
+			if vol.VFrom == "" {
+				vol.PVCName = fmt.Sprintf("%s-claim%d", vol.SvcName, count)
+				count++
+			}
+
+		}
+
+		// We can't assign value to struct field in map while iterating over it, so temporary variable `temp` is used here
+
+		var temp = komposeObject.ServiceConfigs[name]
+		temp.Volumes = FinalVol
+		komposeObject.ServiceConfigs[name] = temp
+		for _, service := range komposeObject.ServiceConfigs {
```

Comment:
why is this for loop in the same kinda for loop? I don't see reason of running this for loop inside https://github.com/kubernetes/kompose/pull/626/files#diff-61f73b3eaf591a7fc27f6c3602535a9aR281

It is running unnecessarily for many iterations.